src/main/java/com/conveyal/datatools/common/utils/S3Utils.java

```diff
@@ -48,30 +35,39 @@
     private static final Logger LOG = LoggerFactory.getLogger(S3Utils.class);
     private static final int REQUEST_TIMEOUT_MSEC = 30 * 1000;
 
-    public static String uploadBranding(Request req, String key) throws IOException, ServletException {
+    public static String uploadBranding(Request req, String key) {
         String url;
 
         String s3Bucket = DataManager.getConfigPropertyAsText("application.data.gtfs_s3_bucket");
         if (s3Bucket == null) {
-            halt(400);
+            haltWith500(
+                req,
+                "s3bucket is incorrectly configured on server",
+                new Exception("s3bucket is incorrectly configured on server")
+            );
         }
 
         // Get file from request
         if (req.raw().getAttribute("org.eclipse.jetty.multipartConfig") == null) {
             MultipartConfigElement multipartConfigElement = new MultipartConfigElement(System.getProperty("java.io.tmpdir"));
             req.raw().setAttribute("org.eclipse.jetty.multipartConfig", multipartConfigElement);
         }
-        Part part = req.raw().getPart("file");
-        String extension = "." + part.getContentType().split("/", 0)[1];
-        File tempFile = File.createTempFile(key + "_branding", extension);
-        InputStream inputStream;
+        String extension = null;
+        File tempFile = null;
         try {
+            Part part = req.raw().getPart("file");
+            extension = "." + part.getContentType().split("/", 0)[1];
+            tempFile = File.createTempFile(key + "_branding", extension);
+            InputStream inputStream;
             inputStream = part.getInputStream();
             FileOutputStream out = new FileOutputStream(tempFile);
             IOUtils.copy(inputStream, out);
-        } catch (Exception e) {
+        } catch (IOException e) {
             e.printStackTrace();
-            haltWithMessage(400, "Unable to read uploaded file");
+            haltWithMessage(req, 400, "Unable to read uploaded file");
+        } catch (ServletException e) {
+            e.printStackTrace();
+            haltWithMessage(req, 400, "Unable to read uploaded file");
```

Comment:
Could you collapse this ServletException clause with the IOException? They appear to do the same thing.

Reply:
Will do